src/Sema.zig

```diff
@@ -30346,11 +30346,6 @@ fn coerceExtra(
         else => {},
     }
 
-    // undefined to anything. We do this after the big switch above so that
-    // special logic has a chance to run first, such as `*[N]T` to `[]T` which
-    // should initialize the length field of the slice.
-    if (maybe_inst_val) |val| if (val.toIntern() == .undef) return pt.undefRef(dest_ty);
-
     if (!opts.report_err) return error.NotCoercible;
```

Comment:
Under `!opts.report_error`, this could cause a valid coercion (`@Type(.undefined)` to anything) to return `error.NotCoercible`, no? I think the undef check should have stayed where it was, but with an extra condition that `dest_ty.toIntern() != .noreturn_type`.

Reply:
I couldn't find any examples of that happening but I guess it might be possible? But wouldn't that break `undefined` to `noreturn` casts? (Is that even desired?)

Reply:
> Under `!opts.report_error`, this could cause a valid coercion (`@Type(.undefined)` to anything) to return `error.NotCoercible`, no? 

I didn't consider that. Alternatively, shouldn't it be okay if the `if (opts.is_ret and dest_ty.zigTypeTag(zcu) == .noreturn)` block gets moved to above the check to coerce `undefined` to anything (i.e. [this change](https://github.com/gabeuehlein/zig/commit/62635a921534af36b950d9195dd4782ca9f1bc21))? That way, `!opts.report_err` will still properly coerce `undefined`, but `coerceExtra` still checks for a return from a `noreturn` function first. It will also avoid having to add the `dest_ty.toIntern() != .noreturn_type` check, which addresses Vexu's problem (if we ever decide that `@as(noreturn, undefined)` is fine).

> But wouldn't that break undefined to noreturn casts?

I'm pretty sure casting anything to `noreturn` is already disallowed right now and is checked before this code path is hit (e.g. `@as(noreturn, undefined)` doesn't compile currently).